### PR TITLE
AST-2428 - SCA Jira tickets severity do not match the priority

### LINKED ISSUE #767: SCA Issues are created in Jira all at default priority
### Description
After configuring priorities for Jira, using the following config:
priorities:
     Critical: Highest
     High: High
     Medium: Medium
     Low: Low
     Informational: Lowest
     
SAST results are properly categorized, all SCA results are default priority for the Jira config, regardless of the SCA severity (High, Medium, Low). 
### Expected Behavior
When a result is High in SCA, the result should be mapped to the high priority in Jira.
### Actual Behavior
SAST results show the correct priority, SCA results are mapped to the default priority (for example, all are Medium).
### Reproduction

1. Run  SAST and SCA scans  with cxflow and Jira configured. 
2. After running the scan, cxflow will open jira tickets on issues
3. Compare SAST issues to SCA issues.
4. SAST Issues will have the proper mapping, SCA issues will all go do default priority. 

### Environment Details

Example Jira Configuration:
jira:
  url:*****
  username: *****
  token:*****
  project:  JAV  
  issue-type: Bug
  priorities:
     Critical: Highest
     High: High
     Medium: Medium
     Low: Low
     Informational: Lowest
  open-transition: In Review
  close-transition: Done
  open-status:
     - To Do
     - In Progress
     - In Review
  closed-status:
     - Done
  fields:
    - type: result
      name: application
      jira-field-name: Application
      jira-field-type: label
    - type: result
      name: cve
      jira-field-name: CVEs
      jira-field-type: label
    - type: result
      name: cwe
      jira-field-name: CWEs
      jira-field-type: label
    - type: result
      name: category
      jira-field-name: Category
      jira-field-type: label
    - type: result
      name: loc
      jira-field-name: LOC
      jira-field-type: label

### PR DESCRIPTION
**Pull request by NimrodGolan**
### Description

> SCA Jira tickets severity do not match the priority

### References

> [Bug URL](https://checkmarx.atlassian.net/browse/AST-2428)

> Fix #767 

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR (if applicable).  *If documentation is a Wiki Update, please indicate desired changes within PR MD Comment*
- [X] All active GitHub checks for tests, formatting, and security are passing
- [X] The correct base branch is being used

### CODE REVIEW

**Comment by cxflowtestuser** [comment]
Scan submitted to Checkmarx

**Comment by cxflowtestuser** [comment]
Logo<br />**Checkmarx SAST - Scan Summary & Details**
### Cx-SAST Summary
Total of 1393 vulnerabilities<br />High&nbsp;**0 High**<br />Medium&nbsp;**0 Medium**<br />Low&nbsp;**1264 Low**<br />Info&nbsp;**129 Info**<br />
### Violation Summary
#### No policy violation found
### Cx-SAST Details
|Lines|Severity|Category|File|Link|
---|---|---|---|---
***
Logo
**Checkmarx SCA - Scan Summary & Details**
### Cx-SCA Summary
**Total Packages Identified**: **269**
**Scan Risk Score**: **9.80**

High&nbsp;**11** **High** **severity vulnerabilities**
Medium&nbsp;**2** **Medium** **severity vulnerabilities**
Low&nbsp;**1** **Low** **severity vulnerabilities**
[View more details on Checkmarx UI](https://eu.sca.checkmarx.net/#/projects/8087bcfd-fba7-4aad-bb0c-9fd2cee68d40/reports/096bf020-d0a5-4f1e-b14c-bc0d3d61c7d6)
### Cx-SCA vulnerability result overview
|Vulnerability ID|Package|Severity|CVSS score|Publish date|Current version|Recommended version|Link in CxSCA|Reference – NVD link|
---|---|---|---|---|---|---|---|---
`CVE-2016-1000031`|commons-fileupload:commons-fileupload|HIGH|9.8|2016-10-25T14:29:00|1.0|| [Vulnerability Link](https://eu.sca.checkmarx.net/#/projects/8087bcfd-fba7-4aad-bb0c-9fd2cee68d40/reports/096bf020-d0a5-4f1e-b14c-bc0d3d61c7d6/vulnerabilities/CVE-2016-1000031%3AMaven-commons-fileupload%3Acommons-fileupload-1.0/vulnerabilityDetails)|[CVE-2016-1000031](https://nvd.nist.gov/vuln/detail/CVE-2016-1000031)
`Cxa9261daf-3755`|org.bouncycastle:bcprov-jdk15on|HIGH|9.8|2021-04-06T11:30:00|1.68|| [Vulnerability Link](https://eu.sca.checkmarx.net/#/projects/8087bcfd-fba7-4aad-bb0c-9fd2cee68d40/reports/096bf020-d0a5-4f1e-b14c-bc0d3d61c7d6/vulnerabilities/Cxa9261daf-3755%3AGradle-org.bouncycastle%3Abcprov-jdk15on-1.68/vulnerabilityDetails)|N\A
`CVE-2016-1000027`|org.springframework:spring-web|HIGH|9.8|2020-01-02T23:15:00|5.2.10.RELEASE|| [Vulnerability Link](https://eu.sca.checkmarx.net/#/projects/8087bcfd-fba7-4aad-bb0c-9fd2cee68d40/reports/096bf020-d0a5-4f1e-b14c-bc0d3d61c7d6/vulnerabilities/CVE-2016-1000027%3AGradle-org.springframework%3Aspring-web-5.2.10.RELEASE/vulnerabilityDetails)|[CVE-2016-1000027](https://nvd.nist.gov/vuln/detail/CVE-2016-1000027)
`CVE-2016-1000027`|org.springframework:spring-webmvc|HIGH|9.8|2020-01-02T23:15:00|5.2.10.RELEASE|| [Vulnerability Link](https://eu.sca.checkmarx.net/#/projects/8087bcfd-fba7-4aad-bb0c-9fd2cee68d40/reports/096bf020-d0a5-4f1e-b14c-bc0d3d61c7d6/vulnerabilities/CVE-2016-1000027%3AGradle-org.springframework%3Aspring-webmvc-5.2.10.RELEASE/vulnerabilityDetails)|[CVE-2016-1000027](https://nvd.nist.gov/vuln/detail/CVE-2016-1000027)
`CVE-2021-22118`|org.springframework:spring-web|HIGH|7.8|2021-05-27T15:15:00|5.2.10.RELEASE|| [Vulnerability Link](https://eu.sca.checkmarx.net/#/projects/8087bcfd-fba7-4aad-bb0c-9fd2cee68d40/reports/096bf020-d0a5-4f1e-b14c-bc0d3d61c7d6/vulnerabilities/CVE-2021-22118%3AGradle-org.springframework%3Aspring-web-5.2.10.RELEASE/vulnerabilityDetails)|[CVE-2021-22118](https://nvd.nist.gov/vuln/detail/CVE-2021-22118)
`CVE-2013-2186`|commons-fileupload:commons-fileupload|HIGH|7.5|2013-10-28T21:55:00|1.0|| [Vulnerability Link](https://eu.sca.checkmarx.net/#/projects/8087bcfd-fba7-4aad-bb0c-9fd2cee68d40/reports/096bf020-d0a5-4f1e-b14c-bc0d3d61c7d6/vulnerabilities/CVE-2013-2186%3AMaven-commons-fileupload%3Acommons-fileupload-1.0/vulnerabilityDetails)|[CVE-2013-2186](https://nvd.nist.gov/vuln/detail/CVE-2013-2186)
`CVE-2016-3092`|commons-fileupload:commons-fileupload|HIGH|7.5|2016-07-04T22:59:00|1.0|| [Vulnerability Link](https://eu.sca.checkmarx.net/#/projects/8087bcfd-fba7-4aad-bb0c-9fd2cee68d40/reports/096bf020-d0a5-4f1e-b14c-bc0d3d61c7d6/vulnerabilities/CVE-2016-3092%3AMaven-commons-fileupload%3Acommons-fileupload-1.0/vulnerabilityDetails)|[CVE-2016-3092](https://nvd.nist.gov/vuln/detail/CVE-2016-3092)
`CVE-2014-0050`|commons-fileupload:commons-fileupload|HIGH|7.5|2014-04-01T06:27:00|1.0|| [Vulnerability Link](https://eu.sca.checkmarx.net/#/projects/8087bcfd-fba7-4aad-bb0c-9fd2cee68d40/reports/096bf020-d0a5-4f1e-b14c-bc0d3d61c7d6/vulnerabilities/CVE-2014-0050%3AMaven-commons-fileupload%3Acommons-fileupload-1.0/vulnerabilityDetails)|[CVE-2014-0050](https://nvd.nist.gov/vuln/detail/CVE-2014-0050)
`CVE-2021-25122`|org.apache.tomcat.embed:tomcat-embed-core|HIGH|7.5|2021-03-01T12:15:00|9.0.39|| [Vulnerability Link](https://eu.sca.checkmarx.net/#/projects/8087bcfd-fba7-4aad-bb0c-9fd2cee68d40/reports/096bf020-d0a5-4f1e-b14c-bc0d3d61c7d6/vulnerabilities/CVE-2021-25122%3AGradle-org.apache.tomcat.embed%3Atomcat-embed-core-9.0.39/vulnerabilityDetails)|[CVE-2021-25122](https://nvd.nist.gov/vuln/detail/CVE-2021-25122)
`CVE-2020-17527`|org.apache.tomcat.embed:tomcat-embed-core|HIGH|7.5|2020-12-03T19:15:00|9.0.39|| [Vulnerability Link](https://eu.sca.checkmarx.net/#/projects/8087bcfd-fba7-4aad-bb0c-9fd2cee68d40/reports/096bf020-d0a5-4f1e-b14c-bc0d3d61c7d6/vulnerabilities/CVE-2020-17527%3AGradle-org.apache.tomcat.embed%3Atomcat-embed-core-9.0.39/vulnerabilityDetails)|[CVE-2020-17527](https://nvd.nist.gov/vuln/detail/CVE-2020-17527)
`CVE-2021-25329`|org.apache.tomcat.embed:tomcat-embed-core|HIGH|7.0|2021-03-01T12:15:00|9.0.39|| [Vulnerability Link](https://eu.sca.checkmarx.net/#/projects/8087bcfd-fba7-4aad-bb0c-9fd2cee68d40/reports/096bf020-d0a5-4f1e-b14c-bc0d3d61c7d6/vulnerabilities/CVE-2021-25329%3AGradle-org.apache.tomcat.embed%3Atomcat-embed-core-9.0.39/vulnerabilityDetails)|[CVE-2021-25329](https://nvd.nist.gov/vuln/detail/CVE-2021-25329)
`CVE-2021-24122`|org.apache.tomcat.embed:tomcat-embed-core|MEDIUM|5.9|2021-01-14T15:15:00|9.0.39|| [Vulnerability Link](https://eu.sca.checkmarx.net/#/projects/8087bcfd-fba7-4aad-bb0c-9fd2cee68d40/reports/096bf020-d0a5-4f1e-b14c-bc0d3d61c7d6/vulnerabilities/CVE-2021-24122%3AGradle-org.apache.tomcat.embed%3Atomcat-embed-core-9.0.39/vulnerabilityDetails)|[CVE-2021-24122](https://nvd.nist.gov/vuln/detail/CVE-2021-24122)
`CVE-2020-17521`|org.codehaus.groovy:groovy|MEDIUM|5.5|2020-12-07T20:15:00|2.5.13|| [Vulnerability Link](https://eu.sca.checkmarx.net/#/projects/8087bcfd-fba7-4aad-bb0c-9fd2cee68d40/reports/096bf020-d0a5-4f1e-b14c-bc0d3d61c7d6/vulnerabilities/CVE-2020-17521%3AGradle-org.codehaus.groovy%3Agroovy-2.5.13/vulnerabilityDetails)|[CVE-2020-17521](https://nvd.nist.gov/vuln/detail/CVE-2020-17521)
`CVE-2013-0248`|commons-fileupload:commons-fileupload|LOW|3.3|2013-03-15T20:55:00|1.0|| [Vulnerability Link](https://eu.sca.checkmarx.net/#/projects/8087bcfd-fba7-4aad-bb0c-9fd2cee68d40/reports/096bf020-d0a5-4f1e-b14c-bc0d3d61c7d6/vulnerabilities/CVE-2013-0248%3AMaven-commons-fileupload%3Acommons-fileupload-1.0/vulnerabilityDetails)|[CVE-2013-0248](https://nvd.nist.gov/vuln/detail/CVE-2013-0248)

**Review by AvivCx — APPROVED** [review]

File: src/main/java/com/checkmarx/flow/service/JiraService.java [hunk 1]
```diff
@@ -358,6 +360,20 @@ private String createIssue(ScanResults.XIssue issue, ScanRequest request) throws
...
+    private String getScannerTypeSeverity(ScanResults.XIssue issue, String severity, List<ScanResults.ScaDetails> scaDetails) {
+        String scannerTypeSeverity;
+        if (ScanUtils.isSAST(issue)) {
+            scannerTypeSeverity = severity;
+        } else {
+            /*
+                Format should be aligned with SAST format
+                In SCA case - HIGH -> High
+             */
+            scannerTypeSeverity = StringUtils.capitalize(Objects.requireNonNull(scaDetails).get(0).getFinding().getSeverity().name().toLowerCase());
```
Comment (veredshahar) [inline src/main/java/com/checkmarx/flow/service/JiraService.java:372]: Need to check if scaDetails can't be null

Comment (NimrodGolan) [reply src/main/java/com/checkmarx/flow/service/JiraService.java:372]: In case SCA scan ends with not results we won't reach this code section

**Pull request merged**